e2e/playwright.config.ts

```diff
@@ -7,19 +7,46 @@ import { defineConfig, devices } from "@playwright/test"
 // import dotenv from 'dotenv';
 // dotenv.config({ path: path.resolve(__dirname, '.env') });
 
+const numWorkers = process.env.CI
+  ? 1
+  : parseInt(process.env.TEST_WORKERS ?? "1")
+
+const workerWebServer = (workerIndex: number) => {
+  const port = 8080 + workerIndex
+  return {
+    command: "./mvnw spring-boot:run",
+    url: `http://127.0.0.1:${port}`,
+    reuseExistingServer: false,
+    cwd: "../server/",
+    env: {
+      SPRING_PROFILES_ACTIVE: "e2e",
+      // per-worker settings (e.g. assign each worker unique DB and port)
+      TEST_WORKER_PORT: port.toString(),
+      TEST_PARALLEL_INDEX: workerIndex.toString(),
+    },
+    /* Pipe server logs to console for troubleshooting */
+    stdout: "pipe",
+    stderr: "pipe",
+  } as const
+}
+
+const workerWebServerConfigs = Array(numWorkers)
+  .fill(0)
```

Comment:
```suggestion
```

Eikö tuo fill ole turha jos heti sen jälkeen teet mapit jossa ignoretat aloitusarvot?

Reply:
TL;DR: Syystä että JavaScript on toisinaan JavaScript, `Array(n).map((_, i) => i)` palauttaa tyhjän taulukon, mutta `Array(n).fill(0).map((_, i) => i)` palauttaa taulukon, jossa on `n` arvoa.

Pidempi, ei-täysin-speksistä-vahvistettu-take-with-a-grain-of-salt selitys: `Array(n)` taitaa vain allokoida oikean kokoisen taulukon, mutta ei lisää taulukkoon mitään, joten taulukon arvot ovat "tyhjiä" (huom. tässä yhteydess **"tyhjä" on eri asia kuin `undefined`**). Jäsä kohtelee taulukollista tyhjiä arvoja samoin kuin tyhjää taulukkoa, eli `.map(..)` ei iteroi tyhjien arvojen yli, jolloin taulukko jää tyhjäksi. Lisäämällä `.fill(..)`, taulukkoon täytetään tyhjien arvojen tilalle jokin aloitusarvo, jolloin myös `.map`:lla on arvoja joiden yli iteroida (koska taulukko ei enää ole tyhjä).

Tuon `n` pituisen taulukon voi rakentaa monella tapaa, `Array(n).fill(..).map(..)` nyt vain sattui olemaan se mihin päädyin tässä tapauksessa.